[webpack] Scheduling app

orangejenny opened the pull request:
## Technical Description

https://dimagi.atlassian.net/browse/SAAS-16273

The "new" libraries here, which haven't been used already on other webpack pages are NVD3 (used on other webpack pages, but this page uses a different version) and ckeditor.

There's a bit of early experimentation in the commit history, but it's still probably easier to review by commit.

## Safety Assurance

### Safety story
Tested locally, and I'll do a bit of testing on staging. As above, most of these dependencies are already in use on webpack pages. CKEditor is behind a flag only used by one project, and also see https://dimagi.atlassian.net/browse/USH-5187

### Automated test coverage

no

### QA Plan

no

### Rollback instructions
- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

orangejenny committed:
Migrated pages depending on ckeditor, switching to 'New Instalaltion Methods'

orangejenny committed:
Renamed scheduling/js/create_schedule.ko

orangejenny committed:
Added ckeditor styles back

I think these were included in the old DLL build.

orangejenny committed:
Migrated broadcasts page

orangejenny committed:
Migrated conditional alerts page to webpack

orangejenny committed:
Migrated messaging dashboard to webpack

orangejenny commented:
Not fixing the pre-existing lint errors on this one.

jingcheng16 commented:
Tried to review but don't feel I have enough knowledge to review this PR... 😢

millerdev commented on corehq/messaging/scheduling/static/scheduling/js/create_schedule.js:
```diff
@@ -1,11 +1,11 @@
     'jquery-ui/ui/widgets/datepicker',
     'bootstrap-timepicker/js/bootstrap-timepicker',
-    'hqwebapp/js/ckeditor_knockout_bindings',
+    'hqwebapp/js/ckeditor_knockout_bindings',       // TODO: only include if feature flag is on
```
Is this TODO expected to be done sometime soon?

orangejenny replied:
Removed this commit.

millerdev commented on corehq/apps/hqwebapp/static/hqwebapp/js/ckeditor_knockout_bindings.js:
```diff
@@ -9,42 +8,45 @@ hqDefine('hqwebapp/js/ckeditor_knockout_bindings', [
     'underscore',
     'knockout',
     'hqwebapp/js/initial_page_data',
+    'ckeditor5/ckeditor5.js',
```
Why does this one have a `.js` extension while others do not?

orangejenny replied:
Lost a comment during an earlier rebase. Added back in https://github.com/dimagi/commcare-hq/pull/35498/commits/218e3c10c81ad1eaa2f1e16fddaa5589939d4337#diff-baa0904f149ab075d59892a3e08593e3e48db2992968850384367903328cdeceR11

There are other `ckeditor.foo` files in the same directory.

millerdev reviewed (commented):
The commit history in this PR is... a bit messy. Apologies for the bluntness, but I'm not sure how to phrase it better right now. Specifically it would be nice if the following commits were either squashed or removed:
- [WIP](https://github.com/dimagi/commcare-hq/pull/35498/commits/68dc636cae0bbe27d40336438a7b0c3ff1934257)
- [Updated to only include ckeditor when feature flag is on](https://github.com/dimagi/commcare-hq/pull/35498/commits/eeffb69809dffab31d0b8e3236ceade39f702a4c)
- [Fixed trailing comma](https://github.com/dimagi/commcare-hq/pull/35498/commits/e69aec109cd8c5d9d5464ac1750f22589e4651fb)
- [Revert "Updated to only include ckeditor when feature flag is on"](https://github.com/dimagi/commcare-hq/pull/35498/commits/a0cc246266bb5e243faafcd30ca454054e7a4f9a)

orangejenny commented:
@millerdev Rebased. Removed the commit that was later reverted, and its revert. Removed/squashed the one-line commits. Squashed the WIP commit into the most related other commit.
